### PR TITLE
Index error empty multilinestring

### PR DESCRIPTION
**Pull request by tomplex**
Add some test cases for empty geometry creation to prove solution of #935.

### CODE REVIEW

**Comment by tomplex** [comment]
Took a minute, but there we go. Not sure why the tests weren't failing before because it wasn't anything I touched, but either way, they're all py2/3 compatible now.

File: tests/test_strtree.py [hunk 1]
```diff
@@ -92,10 +92,11 @@ def test_pickle_persistence():
     """
     tree = STRtree([Point(i, i).buffer(0.1) for i in range(3)])
     pickled_strtree = pickle.dumps(tree)
-    print("pickled strtree:", repr(pickled_strtree))
-    unpickle_script_file_path = Path(__file__).parent / "unpickle-strtree.py"
-    subprocess.run(
+    unpickle_script_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "unpickle-strtree.py")
```
Comment (sgillies) [inline tests/test_strtree.py:95]: Thanks for cleaning up this mess @tomplex . I'm not sure what happened in my merge of #915 but I clearly messed things up in a way that Travis-CI didn't notice.

Comment (tomplex) [reply tests/test_strtree.py:95]: glad I could get it done!

File: tests/test_shape.py [hunk 2]
```diff
@@ -28,3 +29,23 @@ def test_polygon_not_empty_np_array():
...
     assert obj == Polygon([(5, 10), (10, 10), (10, 5)])
+
+
+@pytest.mark.parametrize("geom", [
+    {'type': "MultiLineString", "coordinates": []},
+    {'type': 'MultiLineString', 'coordinates': [[]]},
+    {'type': 'MultiLineString', 'coordinates': None}
+])
+def test_multilinestring_empty(geom):
+    assert shape(geom) == MultiLineString()
```
Comment (jorisvandenbossche) [inline tests/test_shape.py:40]: Should those tests also be added to the master branch? (I understand that this PR targetted maint-1.7, as it was already fixed on master)

Comment (sgillies) [reply tests/test_shape.py:40]: Merged maint-1.7 in 73f4ccb.
